File: Python/priorityQueue/PriorityQueue.py
from People import People

class PriorityQueue:
    def __init__(self):
        self.items = []
        self.length = 0
    
    def empty(self):
        return self.length == 0
    
    def pop(self):
        if not self.empty():
            self.length -= 1
            return self.items.pop(0)

    def show(self):
        for people in self.items:
            print('Nome: %s' % people.getName())
            print('Prioridade: %s' % people.getPriority())

    def push(self, people):
        if self.empty():
            self._push_in_front(people)
        self._push_in_medium(people)

    def _push_in_front(self, people):
        self.items.append(people)
        self.length += 1
        return True
        
    def _push_in_medium(self, people):
        flag_push = False
        for i in range(self.length):
            if self.items[i].getPriority() < people.getPriority():
                self.items.insert(i, people)
                flag_push = True
                break
        if not flag_push:
            self.items.append(people)
        self.length += 1

File: java/src/main/java/org/example/priorityqueue/PriorityQueue.java
package org.example.priorityqueue;

import java.util.ArrayList;
import java.util.List;

public class PriorityQueue {
    private List<People> queue;
    private int length = 0;
    
    public PriorityQueue() {
        this.queue = new ArrayList<>();
    }

    public boolean isEmpty() {
        return length == 0;
    }

    public void pop() {
        if (isEmpty()) {
            return;
        }
        queue.remove(0);
        length--;
    }

    public void show() {
        queue.forEach((people) -> {
            System.out.println(people.getName() + " " + people.getPriority());
        });
    }

    public void push(People people) {
        if (isEmpty()) {
            queue.add(people);
            length++;
            return;
        }
        if (!pushInMedium(people)) {
            queue.add(people);
            length++;
        }
    }

    private boolean pushInMedium(People people) {
        for (int i = 0; i < length; i++) {
            if (queue.get(i).getPriority() < people.getPriority()) {
                queue.add(i, people);
                length++;
                return true;
            }
        }
        return false;
    }
}
